src/subcommand/index_main.cpp

```diff
@@ -1016,6 +1035,33 @@ int main_index(int argc, char** argv) {
 
     }
 
+    //Build snarl distance index
+    if (build_dist) {
+        if (dist_graph.empty() || dist_name.empty()) {
+            cerr << "error: [vg index] distance index requires vg and output file" << endl;
+            return 1;
+        } else {
+            ifstream vg_stream(dist_graph);
+            if (!vg_stream) {
+                cerr << "error: [vg index] cannot open VG file" << endl;
+                exit(1);
+            }
+            VG vg(vg_stream);
+            vg_stream.close();
+
+            CactusSnarlFinder bubble_finder(vg);
```

Comment:
We also have the `vg snarls` command that will serialize snarls to disk. I think it would make sense to also let your command load snarls from one of these files instead of detecting them itself (e.g. [here](https://github.com/vgteam/vg/blob/master/src/subcommand/mpmap_main.cpp#L836)). If you decide to keep both code paths, you should probably also let the snarl manager serialize the snarls it detects too.